xwiki-pro-macros-test/xwiki-pro-macros-test-pageobjects/src/main/java/com/xwiki/pro/test/po/generic/RecentlyUpdatedMacro.java

```diff
@@ -0,0 +1,152 @@
+    }
+
+    public List<String> getItemTitles()
+    {
+        List<WebElement> resultItems = macro.findElements(By.cssSelector(".result-item"));
+        return resultItems.stream().map(item -> item.findElement(By.cssSelector(".result-title a")).getText().trim())
+            .collect(Collectors.toList());
+    }
+
+    public boolean hasShowMoreButton()
+    {
+        return getShowMoreButton().isDisplayed();
+    }
+
+    public void clickShowMore()
+    {
+        getShowMoreButton().click();
+    }
+
+    public boolean hasHeading()
+    {
+        List<WebElement> headings = macro.findElements(By.tagName("h2"));
+        return !headings.isEmpty() && headings.get(0).isDisplayed();
+    }
+
+    public boolean hasAvatars()
+    {
+        List<WebElement> macroResults = macro.findElements(By.cssSelector("img.avatar"));
+        return !macroResults.isEmpty();
+    }
+
+    public String getAuthorName(int resultIndex)
+    {
+
+        List<WebElement> results = macro.findElements(By.cssSelector(".result-item, .result-container"));
+        WebElement result = results.get(resultIndex);
+
+        List<WebElement> authors = result.findElements(By.cssSelector(".result-last-author-name a"));
+        if (!authors.isEmpty()) {
+            return authors.get(0).getText();
+        }
+
+        List<WebElement> plain = result.findElements(By.cssSelector(".result-last-author-name"));
+        if (!plain.isEmpty()) {
+            return plain.get(0).getText();
+        }
+
+        return "";
+    }
+
+    public String getResultsTheme()
+    {
+        WebElement resultsList = macro.findElement(By.cssSelector("ul.results"));
+        String classAttr = resultsList.getAttribute("class");
+
+        for (String cls : classAttr.split(" ")) {
+            if (cls.startsWith("theme-")) {
+                return cls.substring("theme-".length());
+            }
+        }
+
+        return null;
+    }
+
+    public boolean themeStructureIsCorrect(String expectedTheme)
+    {
+        WebElement resultsList = macro.findElement(By.cssSelector("ul.results"));
+
+        List<String> expectedClasses;
+        switch (expectedTheme) {
+            case "concise":
+                expectedClasses = Arrays.asList("result-title", "result-last-author-name", "result-date");
+                break;
+            case "social":
+                expectedClasses = Arrays.asList("result-last-author-name", "result-title", "result-date");
+                break;
+            case "sidebar":
+                expectedClasses = Arrays.asList("result-title", "result-date");
+                break;
+            default:
+                return false;
+        }
+
+        List<WebElement> items = resultsList.findElements(By.cssSelector("li"));
+        for (WebElement item : items) {
+            String itemHtml = item.getAttribute("innerHTML");
+            for (String expectedClass : expectedClasses) {
+                if (!itemHtml.contains(expectedClass)) {
```

Comment:
Are you also checking order with this? Since between concise and social I see order makes a difference